engine/src/java/com/runrev/android/Engine.java

```diff
@@ -3368,6 +3378,11 @@ public void onAppLaunched()
 
 	public void onPause()
 	{
+		for (int i = 0; i < m_lifecycle_listeners.size(); i++)
```

Comment:
Okay so I think we should pause in reverse order. If two components are not related then it doesn't matter.If component B depends on component A, then component A will have been initialized before component B. Therefore A can't care about B, but B might care about A, so B should pause before A.

i.e. Pause in reverse order of registration.